src/renderer/drawers/Provider.js

```diff
@@ -0,0 +1,37 @@
+/* @flow */
+import React, { useReducer } from "react";
+
+type State = {
+  Component: *,
+  props: *,
+  open: boolean,
+};
+
+// actions
+// it makes them available and current from connector events handlers
+export let setDrawer: (Component?: *, props?: *) => void = () => {};
+
+// reducer
+const reducer = (state: State, update) => {
+  return {
+    ...state,
+    ...update,
+  };
+};
+const initialState = {
+  Component: null,
+  props: null,
+  open: false,
+};
+
+export const context = React.createContext<State>(initialState);
+
+const DrawerProvider = ({ children }: { children: React$Node }) => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  setDrawer = (Component, props) => dispatch({ Component, props, open: !!Component });
+
+  return <context.Provider value={state}>{children}</context.Provider>;
```

Comment:
you could expose setDrawer directly in the context something like that
```js 
value={{ ...state, setDrawer }}
```
Its just the fact of exposing mutable functions works in this case but can also be prone to runtime errors.
Thats what my previous comments were about sorry if that wasn't clear enough :sweat_smile: 
Also setDrawer is set each time the provider updates :shrug: if you really wish to preserve this way of doing it try and put the setDrawer = function inside a useEffect at mount maybe

Reply:
mmh i would not recommend to do `value={{ ...state, setDrawer }}` because it makes triggering a redraw each time DrawProvider itself get re-renderer. However the pattern at stake here is a bit weird to me too. There are many possibility indeed. many we should discuss in a quick brainstorm of what we try to achieve and what could be the best.

- your solution is one
- redux is one
- using a event emitter OR a rxjs Subject would be one (to get the data in, then you .subscribe to it)